cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
----
-title: Content Security Policy Cheat Sheet
-permalink: /Content_Security_Policy_Cheat_Sheet/
----
+# Introduction
 
-Content Security Policy (CSP) is an important standard by the W3C that is aimed to prevent a broad range of content injection attacks such as cross-site scripting (XSS).
+This article brings forth a way to integrate the `defense in depth` concept to the client-side of web applications. By injecting the Content-Security-Policy (CSP) headers from the server, the browser is aware and capable of protecting the user from dynamic calls that will load content into the page currently being visited.
 
-Introduction
-============
+## Context
 
-Content Security Policy (CSP) is an effective "defense in depth" technique to be used against content injection attacks. It is a declarative policy that informs the user agent what are valid sources to load from.
+The increase in XSS and clickjacking vulnerabilities demands a more `defense in depth` security approach. CSP comes in place to enforce the loading of resources (scripts, images, etc.) from restricted locations that are trusted by the server, as well as enforcing HTTPS usage transparently. Moreover, the developer will get more visibility on the attacks occurring on the application by using the CSP reporting directive.
 
-Since, it was introduced in Firefox version 4 by Mozilla, it has been adopted as a standard, and grown in adoption and capabilities.
+## Avoid CSP
 
-This document is meant to provide guidance on how to utilize CSP under a variety of situations to address a variety of concerns.
+If you are a developer of any of the applications mentioned below, CSP will barely provide or improve their security:
+- Single page applications with no cookies or authentication.
+- Applications that are already vulnerable to XSS vulnerabilities and chose not to remediate them. CSP is not the first line of defense.
```

Comment:
Not agree: In the context in which you have XSS issue but you cannot fix it (ex: the vendor do not want to fix it) CSP is one of the possible severals lines of defense as well as a WAF...

Reply:
It makes sense under such conditions, in which we are targeting web server admins and not the developers. I didn't look at it from that scope.

Reply:
We can mention that the CSP can be used to try to mitigate exploitation (like the WAF) in a context in which XSS cannot be fixed. Using this way, it give a hint for the sitation that i have taken as example.